RedHatInsights/rhsm-subscriptions
# SWATCH-587: update instance API to filter by hardwareMeasurementType

https://issues.redhat.com/browse/SWATCH-587

Test Steps:

- Create test hosts:
  - `bin/insert-mock-hosts   --num-guests=4   --num-hypervisors=1   --hypervisor-id=eb0d994a-e129-4a18-860d-3e39294f95df`
  - `bin/insert-mock-hosts   --num-aws=1`
- Start app: `DEV_MODE=true ./gradlew :bootRun`
- Run the following 3 curl commands and check that only one instance is returned with the correct category.
```
curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/instances/products/RHEL?report_category=physical' \
  -H 'accept: application/json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```
```
curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/instances/products/RHEL?report_category=cloud' \
  -H 'accept: application/json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```
```
curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/instances/products/RHEL?report_category=virtual' \
  -H 'accept: application/json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```
